cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/SourceControlManagementHttpHandler.java

```diff
@@ -104,6 +130,82 @@ public void deleteRepository(FullHttpRequest request, HttpResponder responder,
                                                               namespaceId));
   }
 
+  /**
+   * Pushes a set of applications configs to linked repository in Json format. It expects a post body that has an
+   * optional commit message and an array of objects, with each object specifying the application and version.
+   * E.g.
+   *
+   * <pre>
+   * {@code
+   * {
+   *   "apps": [
+   *   {"application":"XYZ", "version": "1.23"},
```

Comment:
We need to take versions because:
1. Currently for batch application scan/read, we have to provide full keys of namespace+appName+version, current table spi does not support multiScan of `latest version`, especially that pre 6.8 applications have `latest==NULL`
2. When we write back the sourcecontrol metadata (now it's mainly fileHash) in table, we could not write to the `latest` version since there might be new deployment in between. We need to write back to the pushed versions

Reply:
@albertshau sorry for the confusion. It was more like an update for the API design doc, which originally talks about taking a list of appName(for pushing the latest versions). I am here reasoning why that is not yet achievable in a batch fashion without taking the versions. As you pointed out, it should be a multiRead of N application keys (namespace+appName+version)

Reply:
Might be a corner case, so we will not be able to push version=null pipelines?

Reply:
currently no in this batch push api

Reply:
+1  From API standpoint we should not allow users to push arbitrary versions out of order, this would create a mess where chronology in git won't reflect chronology in CDAP.

Reply:
updated to push latest version only